### PR TITLE
Integrated the new proxy implementation with finality check

### PR DESCRIPTION
**Pull request by iulianpascalau**
- integrated the new proxy implementation with finality check
- changed a return error in eth client wrapper + added unit tests

### CODE REVIEW

**Review by dragos-rebegea — COMMENTED** [review]
Add a proper release for sdk-go after merging [#62](https://github.com/ElrondNetwork/elrond-sdk-erdgo/pull/62)

**Comment by iulianpascalau** [comment]
> Add a proper release for sdk-go after merging [#62](https://github.com/ElrondNetwork/elrond-sdk-erdgo/pull/62)

done

**Review by AdoAdoAdo — APPROVED** [review]

File: config/config.go [hunk 1]
```diff
@@ -176,6 +178,8 @@ type GeneralNotifierConfig struct {
 	MinResultsNum                int
 	PollIntervalInSeconds        uint64
 	AutoSendIntervalInSeconds    uint64
+	ProxyFinalityCheck           bool
```
Comment (bogdan-rosianu) [inline config/config.go:181]: might move the bool param to the bottom for memory alignment purposes. also for `ElrondConfig`

File: clients/ethereum/client_test.go [hunk 2]
```diff
@@ -204,6 +204,60 @@ func TestClient_GetBatch(t *testing.T) {
...
+			},
+			GetBatchDepositsCalled: func(ctx context.Context, batchNonce *big.Int) ([]contract.Deposit, error) {
+				return nil, expectedErr
+			},
+		}
+		batch, err := c.GetBatch(context.Background(), 1)
+		assert.Nil(t, batch)
+		assert.Equal(t, expectedErr, err)
+	})
+	t.Run("deposits mismatch - with 0", func(t *testing.T) {
```
Comment (bogdan-rosianu) [inline clients/ethereum/client_test.go:223]: empty line before a test finish and a new `t.Run` ?

File: config/config.go (hunk 1 above)
Comment (iulianpascalau) [reply config/config.go:181]: done

File: clients/ethereum/client_test.go (hunk 2 above)
Comment (iulianpascalau) [reply clients/ethereum/client_test.go:223]: all tests are written without breaks. In an IDE like goland, the tests are separated by the green run button anyway.

**Review by bogdan-rosianu — APPROVED** [review]

**Review by AdoAdoAdo — APPROVED** [review]

**Pull request merged**